homeassistant/components/homekit_controller/config_flow.py

```diff
@@ -122,6 +128,8 @@ def __init__(self):
 
         _LOGGER.debug("Discovered device %s (%s - %s)", name, model, hkid)
 
+        await self.async_set_unique_id(normalize_hkid(hkid))
+
```

Comment:
You'll want to call `self._abort_if_unique_id_configured()` here.

Reply:
Line 140 should no longer be necessary as no other flows in progress can have the same unique ID.